interface/src/app/gene/network/network.js

```diff
@@ -17,6 +18,23 @@ angular.module('adage.gene.network', [
         controller: 'GeneNetworkCtrl as ctrl'
       }
     },
+    // When "gene_network" state exits, remove "gene-tip" and "edge-tip"
+    // elements from the DOM. (Without this function, the tips window will
+    // be visible in other states too.)
+    onExit: function() {
+      var removeTips = function(className) {
+        var elements = document.getElementsByClassName(className);
+        var i, n;
+        if (elements) { // This should be always true.
+          n = elements.length;
+          for (i = 0; i < n; ++i) {
+            elements[i].parentNode.removeChild(elements[i]);
+          }
+        }
+      };
+      removeTips('gene-tip');
+      removeTips('edge-tip');
+    },
```

Comment:
I know we have to do it this way because the network is done in d3, but we are getting pretty deep into direct DOM manipulation here and that can be tricky to get right when working with Angular. Perhaps the best way to allay that concern would be to have a suite of UI tests that automatically verifies these features are working. I think we will be very thankful to have tests like that whenever we decide to update Angular to a new version.

We are on a tight timeline right now, so it probably doesn't make sense to add in tests to *this* pull request, but perhaps we should file an issue so we remember to do something about it before we contemplate any Angular updates. What do you think?